polarCoordinateTraversal.m

```diff
@@ -169,30 +169,21 @@
         t = tMaxR;
         p = ray_origin + t.*ray_direction;
         r_new = sqrt((p(1) - circle_center(1))^2 + (p(2) - circle_center(2))^2);
-        if (abs(r_new - r) < eps)
-        % Note that tStepR is only used for plotting, not computationally significant. 
-            tStepR = 0; 
-        elseif (r_new - r > 0) 
-            tStepR = -1; 
-            current_voxel_ID_r = current_voxel_ID_r -1;
-        else
-            tStepR = 1; 
-            current_voxel_ID_r = current_voxel_ID_r + 1;
-        end
+        current_voxel_ID_r = current_voxel_ID_r + tStepR;
         r = r_new;
         
         if (current_voxel_ID_r <= 0), return; end
```

Comment:
Also, since we want to exit as soon as we hit the end, we can probably move this right after wherever tStepR is calculated. I had to put this down here initially since we didn't calculate tStepR till pretty late in the loop.